src/App/Material.h

```diff
@@ -131,6 +131,8 @@ class AppExport Material
     Color emissiveColor; /**< Defines the emissive color. */
     float shininess;
     float transparency;
+    std::string image;
+    std::string imagePath;

```

Comment:
What's the reason for adding image/texture path to the Material class? And why are there two strings?
Material and image/texture handling are two separate things and this is an example of breaking the SRP.

Reply:
This is to support textures as part of materials. It's not used yet, but will be once textures are supported.

It can be removed for now but will only be added back later

Reply:
Image is a base64 image, image path is a path to an image file. Only one will be used

Reply:
And what is SRP?

Reply:
"[Single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)" -- though I'm not really sure how it's applicable here, to be honest. Maybe Werner can clarify? It seems to me that the displayed texture (and its file) are pretty central to rendering a material, at least once that feature is implemented.

Reply:
The Material class basically is to store the four colours, transparency and shininess. This information is applied to the SoMaterial node of the OpenInventor (OI) library. The SoMaterial class doesn't know anything about images or textures.

In OI textures are basically handled by SoTexture2 and SoTexture3 and images by SoImage. 
For advanced handling of textures OI has around 20 further classes to control coordinates, transformations, reflections, mapping and many more things.

Sooner or later some of the additional options will be needed to fine-tune the rendering of textures.

The result by having all texture related parameters in the Material class is that all other classes in FC that set or get them will have methods to either only handle the colour parts or the texture parts but not both. So, these methods can be divided into two parts with no cohesion which IMO can be considered as breaking the SRP.

To avoid this situation material and texture handling should be kept strictly separated.